django/api/models/go_electric_rebate_application.py

```diff
@@ -126,6 +128,18 @@ def doc2_tag(self):
     consent_personal = BooleanField(validators=[validate_consent], null=True)
     consent_tax = BooleanField(validators=[validate_consent], null=True)
     reason_for_decline = CharField(max_length=500, unique=False, blank=True, null=True)
+    submission_date = DateTimeField(auto_now_add=True, null=True)
+    approved_on = DateTimeField(null=True, editable=False)
+    not_approved_on = DateTimeField(null=True, editable=False)
+
+    def save(self, *args, **kwargs):
```

Comment:
We use bulk_update to update application statuses to approved/not-approved (please see update_application_statuses() in rebate.py); hence, this save() method won't be called when the application becomes approved or not approved.